lib/libutils/isoc/bget_malloc.c

```diff
@@ -110,16 +110,22 @@
 #if defined(__KERNEL__)
 /* Compiling for TEE Core */
 #include <kernel/asan.h>
-#include <kernel/mutex.h>
+#include <kernel/thread.h>
+#include <kernel/spinlock.h>
 
-static void malloc_lock(void)
+static unsigned int malloc_lock(void)
 {
-	mutex_lock(&__malloc_mu);
+	unsigned int exceptions;
```

Comment:
Please use `uint32_t` instead as that's what `thread_mask_exceptions()` returns.